app/src/main/java/de/westnordost/streetcomplete/util/prefs/Preferences.kt

```diff
@@ -0,0 +1,29 @@
+package de.westnordost.streetcomplete.util.prefs
+
+interface Preferences {
+    interface Listener {
+        fun onPreferencesChanged(key: String)
+    }
+
+    fun putBoolean(key: String, boolean: Boolean)
+    fun putInt(key: String, int: Int)
+    fun putLong(key: String, long: Long)
+    fun putFloat(key: String, float: Float)
+    fun putDouble(key: String, double: Double)
+    fun putString(key: String, string: String?)
+
+    fun getBoolean(key: String, defaultValue: Boolean): Boolean
+    fun getInt(key: String, defaultValue: Int): Int
+    fun getLong(key: String, defaultValue: Long): Long
+    fun getFloat(key: String, defaultValue: Float): Float
+    fun getDouble(key: String, defaultValue: Double): Double
+    fun getString(key: String, defaultValue: String?): String?
```

Comment:
I think it would be nice to have the second parameter optional and have reasonable defaults (documented as part of the signature). E.g. 

```suggestion
    fun getBoolean(key: String, defaultValue: Boolean = false): Boolean
    fun getInt(key: String, defaultValue: Int = 0): Int
    fun getLong(key: String, defaultValue: Long = 0L): Long
    fun getFloat(key: String, defaultValue: Float = 0f): Float
    fun getDouble(key: String, defaultValue: Double = 0.0): Double
    fun getString(key: String, defaultValue: String? = null): String?
```

Not sure about this, but maybe actually make it all nullable (so that default is always `null`)?

You've combed through the source code, what do you think would be the most convenient (and self-explanatory) interface?